domenetjenester/vedtak/src/main/java/no/nav/foreldrepenger/domene/vedtak/migrering/MigrerManglendeForespørslerTjeneste.java

```diff
@@ -0,0 +1,96 @@
+import no.nav.foreldrepenger.domene.fpinntektsmelding.OrganisasjonsnummerDto;
+import no.nav.foreldrepenger.domene.uttak.saldo.StønadskontoSaldoTjeneste;
+import no.nav.foreldrepenger.domene.uttak.saldo.svp.MaksDatoUttakTjenesteImpl;
+import no.nav.foreldrepenger.skjæringstidspunkt.SkjæringstidspunktTjeneste;
+import no.nav.vedtak.konfig.Tid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class MigrerManglendeForespørslerTjeneste {
+    private static final Logger LOG = LoggerFactory.getLogger(MigrerManglendeForespørslerTjeneste.class);
+
+    private BehandlingRepository behandlingRepository;
+    private SkjæringstidspunktTjeneste skjæringstidspunktTjeneste;
+    private StønadskontoSaldoTjeneste stønadskontoSaldoTjeneste;
+    private UttakInputTjeneste uttakInputTjeneste;
+    private FpInntektsmeldingTjeneste fpInntektsmeldingTjeneste;
+    private InntektsmeldingTjeneste inntektsmeldingTjeneste;
+    private MaksDatoUttakTjenesteImpl maksDatoUttakTjenesteSvp;
+
+    public MigrerManglendeForespørslerTjeneste() {
+        // for CDI proxy
+    }
+
+    @Inject
+    public MigrerManglendeForespørslerTjeneste(BehandlingRepository behandlingRepository, SkjæringstidspunktTjeneste skjæringstidspunktTjeneste,
+                                               StønadskontoSaldoTjeneste stønadskontoSaldoTjeneste,
+                                               UttakInputTjeneste uttakInputTjeneste,
+                                               FpInntektsmeldingTjeneste fpInntektsmeldingTjeneste,
+                                               InntektsmeldingTjeneste inntektsmeldingTjeneste,
+                                               @FagsakYtelseTypeRef(FagsakYtelseType.SVANGERSKAPSPENGER) MaksDatoUttakTjenesteImpl maksDatoUttakTjenesteSvp) {
+        this.behandlingRepository = behandlingRepository;
+        this.skjæringstidspunktTjeneste = skjæringstidspunktTjeneste;
+        this.stønadskontoSaldoTjeneste = stønadskontoSaldoTjeneste;
+        this.uttakInputTjeneste = uttakInputTjeneste;
+        this.fpInntektsmeldingTjeneste = fpInntektsmeldingTjeneste;
+        this.inntektsmeldingTjeneste = inntektsmeldingTjeneste;
+        this.maksDatoUttakTjenesteSvp = maksDatoUttakTjenesteSvp;
+    }
+
+    public void migrerManglendeForespørsler() {
+        // implementasjon
+    }
+
+    public void vurderOmForespørselSkalOpprettes(Fagsak sak, boolean dryRun) {
+        var stp = skjæringstidspunktTjeneste.getSkjæringstidspunkter(sak.getId());
+        var sisteYtelsebehandling = behandlingRepository.finnSisteIkkeHenlagteYtelseBehandlingFor(sak.getId());
+
+        sisteYtelsebehandling.ifPresent(behandling -> {
+            var uttakInput = uttakInputTjeneste.lagInput(behandling);
+            if (FagsakYtelseType.FORELDREPENGER.equals(behandling.getFagsak().getYtelseType())) {
```

Comment:
Hm, hvilke tilfelle ønsker du migrere? 
Hvis du tenker varig lønnsendring fom 1/5-2024 (de blir etterbetalt for oss i mars / april), er det ikke da like greit å sende de med STP >= den datoen?

Reply:
ikke helt sikker på hva som blir det riktigste intervallet her...når det sendes inn en agi så har vi en kontroll på om det finnes eksisterende sak, hvis ikke får arbeidsgiver ikke sendt inn...